src/ApplicationInsights.Kubernetes/K8sInfoBackgroundService.cs

```diff
@@ -30,6 +30,26 @@ internal class K8sInfoBackgroundService : BackgroundService
         _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
     }
 
+    // Allows non-hosted service to bootstrap cluster info fetch.
+    void IK8sInfoBootstrap.Run()
+    {
+        try
+        {
+            // Fire and forget on purpose to avoid blocking the client code's thread.
+            _ = Task.Run(async () =>
+            {
+                using (CancellationTokenSource cancellationTokenSource = new CancellationTokenSource())
+                {
+                    await ExecuteAsync(cancellationTokenSource.Token).ConfigureAwait(false);
```

Comment:
`CancellationToken.None`?

Or consider passing in a `CancellationToken` and allowing to cancel the bootstrapping process

Reply:
Let me see if None is an option since you pointed it out.
I didn't pay too much effort there because I think that are implementation details and we can always improve later.

Reply:
The ability to cancel bootstrapping (if introduced) would be a contract change, not just implementation detail. Also, ability to shut down gracefully and timely is generally very desirable.

Reply:
Used `default` to supply an empty cancellation token for this scenario.